src/sidebar/services/router.js

```diff
@@ -94,9 +98,15 @@ export class RouterService {
         throw new Error(`Cannot generate URL for route "${name}"`);
     }
 
-    const query = queryString.stringify(queryParams);
-    if (query.length > 0) {
-      url += '?' + query;
+    let hasParams = false;
+    const searchParams = new URLSearchParams();
+    for (let [key, value] of Object.entries(queryParams)) {
```

Comment:
The `params` parameter object gets copied to `queryParams` and then is iterated over to build `searchParams` (eyes mildly crossing). Not that I have a better idea; there are just a lot of references with "params" in their name flying around here...